app/controllers/api/v1/todos_controller.rb

```diff
@@ -0,0 +1,43 @@
+class Api::V1::TodosController < ApiController
+  before_action :restrict_access   
+
+  def index
+    todos = @current_user.todos
+    render json: todos
+
+  end                          
+
+  def create
+    todo = @current_user.todos.create(todo_params)
+    if todo.save
```

Comment:
Lines 11 and 12 repeat the same work of saving to the database.

`@current_user.todos.create(...)` on line 11 will save the todo to the database if its valid. 

Line 12 currently will also try to save the todo to the database. 

Saving to the database only needs to happen *once* in the action.

Instead of `create` on line 11, use `build`. `build` does not try to save the object to the database.